stores/slabbuffer.go

```diff
@@ -278,10 +278,22 @@ func (mgr *SlabBufferManager) AddPartialSlab(ctx context.Context, data []byte, m
 			return nil
 		}()
 		if err != nil {
-			return nil, fmt.Errorf("failed to update size/complete in db: %w", err)
+			return nil, 0, fmt.Errorf("failed to update size/complete in db: %w", err)
 		}
 	}
-	return slabs, nil
+
+	return slabs, mgr.BufferSize(gid), nil
+}
+
+func (mgr *SlabBufferManager) BufferSize(gid bufferGroupID) (total int64) {
+	mgr.mu.Lock()
+	defer mgr.mu.Unlock()
+	for _, buffer := range append(mgr.completeBuffers[gid], mgr.incompleteBuffers[gid]...) {
```

Comment:
should be fine I think, wanted to draw attention to it nonetheless cc @ChrisSchinnerl

Reply:
General rule of thumb is that you shouldn't use `append` to create a new slice and instead only if you wish to actually append to an existing slice since it manipulates the underlying slice which can lead to hard-to-catch errors.
So I'd go the safe route here and use 2 loops.